### ISSUE TITLE
Better programatic support for adding accounts from known keys

### ISSUE THREAD

**Issue by unparalleled-js** [ApeWorX/ape#1102]
### Overview

Feature request from a user:

Allow hardcoded accounts in tests

### Specification

```python
class HardcodedAccountContainer(AccountContainerAPI):
    def create_accounts(self, public_key: str, private_key: str) -> HardcodedAccount:
        ...
```

And implement `HardcodedAccount` much like the other accounts.

### Dependencies

Include links to any open issues that must be resolved before this feature can be implemented.

**Comment by fubuloubu** [comment]
We have `AccountContainerAPI.append(account: AccountAPI)` to add new accounts to containers

Also, separately, what about `ape accounts import [alias]`?

**Comment by unparalleled-js** [comment]
> Also, separately, what about `ape accounts import [alias]`?

The user requesting this feature wants to do it more programmatically without using `subprocess.run()`.

**Comment by fubuloubu** [comment]
> > Also, separately, what about `ape accounts import [alias]`?
> 
> The user requesting this feature wants to do it more programmatically without using `subprocess.run()`.

Can probably re-write some of that CLI endpoint to make more of it exposed to programming interface e.g. `KeyfileAccount.from_private_key(key: bytes)` and then have it write the json file there so it gets recognized by the `KeyfileAccountContainer`

Or did this user want to have "temporary" accounts for the purposes of testing or something?

**Comment by unparalleled-js** [comment]
I just realized this almost works:

```python
from ape_test.accounts import TestAccount

acct = TestAccount(index=11, public_key="0x...", private_key="0x0...")
```

It would be nice to make `index` optional but I think this could unblock our user

**Comment by PCanavelli** [comment]
Hi everyone - I'm the user who submitted this feature request on Discord.

The use case is basically being able to programmatically instantiate ape accounts from within a script and use them to sign transactions without using the CLI, assuming the public and private keys are locally available. 

It's all for testing purposes using throwaway wallets at the moment, so security concerns can be neglected. But I'd appreciate any feedback as to how this workflow could transfer to main nets with actual currencies at stake.

Also, @unparalleled-js's `TestAccount` fix looks exactly like what I was after.

**Comment by NotPeopling2day** [comment]
Superseded by https://github.com/ApeWorX/ape/pull/1199
